### ISSUE TITLE
[bug] profile ``[tool_requires]`` revisions are being ignored

### ISSUE THREAD

**Issue by thorsten-klein** [conan-io/conan#14327]
### Environment details

* Operating System+version: Ubuntu20
* Compiler+version: irrelevant
* Conan version: 2.0.7
* Python version: 3.8.10


### Steps to reproduce

File myprofile:
```
[tool_requires]
*:mytool/1.0.0@myowner/mychannel#e8e2d38fb0887c3b6ccc8b04bf697002
```

Then I run: `conan graph info . -pr:b=myprofile`:

It seems that the RREV is not taken into account, and another RREV is downloaded.

### Logs

`conan graph info . -pr:b=myprofile`:

```
======== Computing dependency graph ========                                                                                                                                                                                                                 
mytool/1.0.0@myowner/mychannel: Not found in local cache, looking in remotes...
mytool/1.0.0@myowner/mychannel: Checking remote: myremote
mytool/1.0.0@myowner/mychannel: Downloaded recipe revision 88d6c60b4d4fa25cbce81f3495f2855a
```

**Comment by memsharded** [comment]
Thanks very much for reporting @thorsten-klein 

Indeed a bug, I am proposing a fix in https://github.com/conan-io/conan/pull/14337 for next 2.0.10 release.

**Comment by memsharded** [comment]
https://github.com/conan-io/conan/pull/14337 merged, it will be in next 2.0.10
